mindmeld/text_preparation/normalizers.py

```diff
@@ -0,0 +1,222 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2015 Cisco Systems, Inc. and others.  All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module contains Normalizers."""
+import unicodedata
+import codecs
+from abc import ABC, abstractmethod
+import logging
+
+from ..constants import ASCII_CUTOFF
+from ..path import ASCII_FOLDING_DICT_PATH
+
+logger = logging.getLogger(__name__)
+
+
+class Normalizer(ABC):
+    """Abstract Normalizer Base Class."""
+
+    def __init__(self):
+        """Creates a Normalizer instance."""
+
+    @abstractmethod
+    def normalize(self, text):
+        """
+        Args:
+            text (str): Input text.
+        Returns:
+            normalized_text (str): Normalized Text.
+        """
+        raise NotImplementedError("Subclasses must implement this method")
+
+
+class NoOpNormalizer(Normalizer):
+    """A No-Ops Normalizer."""
+
+    def normalize(self, text):
+        """
+        Args:
+            text (str): Input text.
+        Returns:
+            normalized_text (str): Returns the original text.
+        """
+        return text
+
+
+class ASCIIFold(Normalizer):
+    """An ASCII Folding Normalizer."""
+
+    def __init__(self):
+        super().__init__()
+        self.ascii_folding_table = self.load_ascii_folding_table()
+
+    def normalize(self, text):
+        """
+        Args:
+            text (str): Input text.
+        Returns:
+            normalized_text (str): Normalized Text.
+        """
+        return self.fold_str_to_ascii(text)
+
+    def fold_char_to_ascii(self, char):
+        """
+        Return the ASCII character corresponding to the folding token.
+
+        Args:
+            char: ASCII folding token
+
+        Returns:
+            char: a ASCII character
+        """
+        char_ord = ord(char)
+        if char_ord < ASCII_CUTOFF:
```

Comment:
Checking if this should be `if char_ord <= ASCII_CUTOFF:`

Reply:
The cut off of `\u0080` is hex for 128 and since ASCII only goes up to 127 the current check is correct. 

https://www.garykessler.net/library/ascii.html